Sources/Sentry/Profiling/SentryProfiledTracerConcurrency.mm

```diff
@@ -170,21 +168,15 @@
 }
 
 void
-sentry_discardProfilerCorrelatedToTrace(SentryId *internalTraceId, SentryHub *hub, BOOL isProfiling)
+sentry_discardProfilerCorrelatedToTrace(SentryId *internalTraceId, SentryHub *hub)
 {
     std::lock_guard<std::mutex> l(_gStateLock);
 
     if ([SentryContinuousProfiler isCurrentlyProfiling]) {
-        BOOL notLaunchContinuousProfileV2TraceLifecycle = hub != nil
-            && hub.client.options.profiling != nil
-            && hub.client.options.profiling.lifecycle == SentryProfileLifecycleTrace;
-        if (!notLaunchContinuousProfileV2TraceLifecycle && !isProfiling) {
-            SENTRY_LOG_DEBUG(@"Continuous profiler v1 won't be stopped with a tracer.");
-            return;
-        }
-
+        SENTRY_LOG_DEBUG(@"Stopping tracking discarded tracer with profileReferenceId %@",
+            internalTraceId.sentryIdString);
         _unsafe_cleanUpContinuousProfilerV2();
-    } else if (internalTraceId != nil && isProfiling) {
+    } else if (internalTraceId != nil) {
```

Comment:
Two things were wrong here, one is that hub is nil for launch profiles, so the condition checked on line 181 was wrong. I also removed the `isProfiling` parameter and only call this function from SentryTracer if it is `true`.